File: multiples-of-3-or-5/solution.py
def solution4(number):
    l = [x for x in range(number) if not x%3 or not x%5]
    return sum(l)

def solution(number):
    return sum(x for x in range(number) if not x%3 or not x%5)

print(solution(4))
print(solution(6))
print(solution(16))
print(solution(3))
print(solution(5))

File: multiples-of-3-or-5/solution.hs
module MultiplesOf3And5 where

solution :: Integer -> Integer
solution number = sum [n | n <- [1..number - 1], n `mod` 3 == 0 || n `mod` 5 == 0]

main :: IO ()
main =
    do
        putStrLn . show $ solution 16
        putStrLn (show (solution 16 ))